internal/quota/resources.go

```diff
@@ -28,10 +28,15 @@ func ResourceListEqual(resourceListA, resourceListB corev1.ResourceList) bool {
 }
 
 // ResourceQuotaSpecEqual gets two ResourceQuotaSpecs and returns whether their specs are equal.
-func ResourceQuotaSpecEqual(resourceQuotaSpecA, resourceQuotaSpecB corev1.ResourceQuotaSpec) bool {
+func ResourceQuotaSpecEqual(resourceQuotaSpecA, resourceQuotaSpecB corev1.ResourceQuotaSpec, snsObject *objectcontext.ObjectContext) bool {
 	var resources []string
 
-	for resourceName := range ZeroedQuota.Hard {
+	configMapData, err := GetConfigMapData(snsObject.Ctx, snsObject.Client)
+	if err != nil {
+		return false
+	}
```

Comment:
Change the `ResourceQuotaSpecEqual` function signature to use `ResourceQuotaSpec` instead of getting the `configMap` in the function.